src/Database.h

```diff
@@ -0,0 +1,50 @@
+#ifndef _DATABASE_H_
+#define _DATABASE_H_
+
+#include "SURLCreator.h"
+#include <unordered_map>
+#include <mutex>
+
+// Database save the look-up table to link each SURL to there original URL
+// public implement: create, query, insert
```

Comment:
How about defining it based on CRUD?

https://zh.wikipedia.org/wiki/%E5%A2%9E%E5%88%AA%E6%9F%A5%E6%94%B9